terraform/main.tf

```diff
@@ -301,6 +332,213 @@ resource "aws_iam_role_policy" "api_task-publish_to_arpa_audit_report_queue" {
+    access_point_id = module.api.efs_data_volume_access_point_id
+  }]
+  # Task resource configuration
+  # TODO: Tune these values after observing usage in different environments.
+  #       See also: --max_old_space_size in NODE_OPTIONS env var.
+  consumer_task_size = {
+    cpu    = 1024 # 1 vCPU
+    memory = 4096 # 4 GB
+  }
+
+  # Messaging
+  autoscaling_message_thresholds = [1, 3, 5, 10, 20, 50]
+  sqs_publisher = {
+    principal_type       = "Service"
+    principal_identifier = "ecs-tasks.amazonaws.com"
+    source_arn           = module.api.ecs_service_arn
+  }
+  sqs_max_receive_count             = 2
+  sqs_dlq_message_retention_seconds = 1209600 # 14 days, in seconds
+
+  # Logging
+  log_retention = var.api_log_retention_in_days
+
+  # Secrets
+  ssm_path_prefix = var.ssm_service_parameters_path_prefix
+
+  # Postgres
+  rds_db_connect_resources = module.postgres.rds_db_connect_resources_list
+  postgres_username        = module.postgres.master_username
+  postgres_endpoint        = module.postgres.cluster_endpoint
+  postgres_port            = module.postgres.cluster_port
+  postgres_db_name         = module.postgres.default_db_name
+}
+
+data "aws_iam_policy_document" "publish_to_digest_email_send_queue" {
+  statement {
+    sid       = "AllowPublishToQueue"
+    actions   = ["sqs:SendMessage"]
+    resources = [module.digest_email_send.sqs_queue_arn]
+  }
+}
+
+resource "aws_iam_role_policy" "digest_email_get_grants-publish_to_digest_email_send_queue" {
+  name_prefix = "send-digest-email-send-requests"
+  role        = module.digest_email_get_grants.ecs_task_role_name
+  policy      = data.aws_iam_policy_document.publish_to_digest_email_send_queue.json
+}
+
+module "digest_email_send" {
+  source                   = "./modules/sqs_consumer_task"
+  namespace                = "${var.namespace}-digest_email_send"
+  permissions_boundary_arn = local.permissions_boundary_arn
+
+  # Networking
+  subnet_ids         = local.private_subnet_ids
+  security_group_ids = [module.digest_email_send_security_group.id]
+
+  # Task configuration
+  ecs_cluster_name      = join("", aws_ecs_cluster.default.*.name)
+  docker_tag            = var.api_container_image_tag
+  unified_service_tags  = local.unified_service_tags
+  stop_timeout_seconds  = 120
+  consumer_task_command = ["node", "./src/lib/digest/sendEmail.js"]
+  consumer_container_environment = {
+    API_DOMAIN          = local.api_domain_name
+    DATA_DIR            = "/var/data"
+    NODE_OPTIONS        = "--max_old_space_size=3584" # Reserve 512 MB for other task resources
+    NOTIFICATIONS_EMAIL = "grants-notifications@${var.website_domain_name}"
+    WEBSITE_DOMAIN      = "https://${var.website_domain_name}"
+  }
+  consumer_task_efs_volume_mounts = [{
+    name            = "data"
+    container_path  = "/var/data"
+    read_only       = false
+    file_system_id  = module.api.efs_data_volume_id
+    access_point_id = module.api.efs_data_volume_access_point_id
+  }]
```

Comment:
Same as with the previous task – I think this can be removed.
```suggestion
```

Reply:
I initially removed it, but ran into a terraform validation error as it seemed to be expecting this volume mount to exist. Let me try again.

https://github.com/usdigitalresponse/usdr-gost/actions/runs/6397702168/job/17366227447

```
Error: Missing required argument

  on main.tf line 471, in module "digest_email_send":
 471: module "digest_email_send" {

The argument "consumer_task_efs_volume_mounts" is required, but no definition
was found.
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.
```

Reply:
Ah, good find. We discussed this in our call today, but just to state for the record: you're running into this because of the [variable definition in the module](https://github.com/usdigitalresponse/usdr-gost/blob/_staging/terraform/modules/sqs_consumer_task/variables.tf#L38-L47), here:
```tf
variable "consumer_task_efs_volume_mounts" {
  description = "EFS volumes to mount in ECS task consumer containers."
  type = list(object({
    name            = string
    container_path  = string
    read_only       = bool
    file_system_id  = string
    access_point_id = string
  }))
}
```

The module uses this variable in two separate `for` loops to configure a separate volume mount for each list item provided to this variable when "calling" the module. Moreover, Terraform sees this particular variable as a required input because it doesn't have a `default` value assignment. 

So, there are two ways to fix:
1. When calling the module, provide an empty list as the value for this input variable, e.g.
    ```tf
    consumer_task_efs_volume_mounts = []
    ```
2. Set a `default` value of an empty list in the variable definition lined above:
    ```tf
    variable "consumer_task_efs_volume_mounts" {
      description = "EFS volumes to mount in ECS task consumer containers."
      type = list(object({
        name            = string
        container_path  = string
        read_only       = bool
        file_system_id  = string
        access_point_id = string
      }))
      default = []
    }
    ```

Given that an empty list works just fine, I don't think it should be a requirement in the first place, so I'd advocate for the second option.